### PR TITLE
[rush-archive-project-plugin]: Add md file for archived projects

### PR DESCRIPTION
**Pull request by william2958**
### Basic Checks

Have you run `rush change` for this change?

- [x] Yes
- [ ] No

If **No**, please run `rush change` before, this is necessary.

If adding a **new feature**, the PR's description includes:

- [ ] Reason for adding this feature
- [ ] How to use
- [ ] A basic example

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing applications:

### Summary
This PR adds the following changes:
- Query user if they would like to push the checkpoint to remote
- Save the archived project details json to the root graveyard folder
- Generate a MD file for better user readability next to the archived project details json file

### CODE REVIEW

File: common/autoinstallers/command-plugins/package.json [hunk 1]
```diff
@@ -3,7 +3,7 @@
   "version": "1.0.0",
   "private": true,
   "dependencies": {
-    "rush-archive-project-plugin": "1.1.3",
+    "rush-archive-project-plugin": "link:../../../rush-plugins/rush-archive-project-plugin",
```
Comment (chengcyber) [inline common/autoinstallers/command-plugins/package.json:6]: We need to remove this before merge

File: rush-plugins/rush-archive-project-plugin/src/commands/archive.ts [hunk 2]
```diff
@@ -55,8 +58,16 @@ ${consumingProjectNames.join(", ")}`);
     spinner = ora('Attempting to create a git checkpoint branch');
     const branchName: string = getCheckpointBranch(rushConfiguration.rushJsonFolder,packageName);
     spinner.succeed(`Git Checkpoint created at branch: ${branchName}`);
+    // Prompt user to push the newly created branch
+    const { pushBranch } = await inquirer.prompt([
```
Comment (chengcyber) [inline rush-plugins/rush-archive-project-plugin/src/commands/archive.ts:62]: Beside used as a command line tool, this `archive` function may be also invoked as Node API. So, if introduced a new prompt here, it hangs when used as Node API usage.


I can see there are different solutions:

1. move the prompt in `cli.ts` and do the rest of logic
2. Or, Add a `shouldPushGitCheckpoint` option for `archive` function, and skip the prompt here if the option passed

Comment (william2958) [reply rush-plugins/rush-archive-project-plugin/src/commands/archive.ts]: I added the shouldPushGitCheckpoint option, is there a node api call I need to modify as well?

File: common/autoinstallers/command-plugins/package.json (hunk 1 above)
Comment (william2958) [reply common/autoinstallers/command-plugins/package.json]: removed

**Review by chengcyber — APPROVED** [review]

**Pull request merged**